linera-core/src/value_cache.rs

```diff
@@ -164,38 +168,23 @@ where
     }
 }
 
-impl ValueCache<CryptoHash, HashedCertificateValue> {
+impl<T: Clone> ValueCache<CryptoHash, Hashed<T>> {
     /// Inserts a [`HashedCertificateValue`] into the cache, if it's not already present.
     ///
     /// The `value` is wrapped in a [`Cow`] so that it is only cloned if it needs to be
     /// inserted in the cache.
     ///
     /// Returns [`true`] if the value was not already present in the cache.
-    ///
-    /// # Notes
-    ///
-    /// If the `value` is a [`HashedCertificateValue::ValidatedBlock`], its respective
-    /// [`HashedCertificateValue::ConfirmedBlock`] is also cached.
-    pub async fn insert<'a>(&self, value: Cow<'a, HashedCertificateValue>) -> bool {
+    pub async fn insert<'a>(&self, value: Cow<'a, Hashed<T>>) -> bool {
```

Comment:
Want to emphasize that we no longer insert a validated variant of the hashed value if the confirmed one is being inserted.

Reply:
(It was the other way round, I think?)
If I'm reading this correctly we _are_ inserting the confirmed variant when we _sign_ it; that should still take care of the most common case, so that most validators that receive a lite confirmed block certificate will already have its value in the cache.